doc/crypto/api/ops/encap.rst

```diff
@@ -0,0 +1,326 @@
+Key Encapsulation
+=================
+
+Elliptic Curve Integrated Encryption Scheme
+-------------------------------------------
+
+The Elliptic Curve Integrated Encryption Scheme was fist proposed by Shoup, then improved by Ballare and Rogaway.
+
+The original specification permitted a number of variants. This specification only defines the version specified in [SEC1], that is with the use of labels and with the label size defined in bytes. 
+
+It is possible that some applications may need to use older versions to interoperate with legacy systems. 
+While the application can always implement this using the other algorithm functions provided, however, an implementation may choose to add these as a convenience in the implementation numbering space. 
+
+.. macro:: PSA_ALG_ECIES
+    :definition: ((psa_algorithm_t)0x09yyyxxxx)
+
+    .. summary::
+       The Elliptic Curve Integrated Encryption Scheme.
+
+       This algorithm can only be used when combined with a key derivation operation using `PSA_ALG_ENCAPSULATION()` in a call to `psa_encapsulate_key()`
+
+       When used as a key's permitted-algorithm policy, the following uses are permitted:
+
+       *   In a call to `psa_encapsulate_key()` or `psa_decapsulate_key()`, with any combined key establishment and key derivation algorithm constructed with `PSA_ALG_ECIES`.
+
+       This encapsulation scheme is defined by :cite-title:`SEC1` �5.5.1 under the name Elliptic Curve Integrated Encryption Scheme.
+
+       .. subsection:: Compatible key types
+
+           | :code:`PSA_KEY_TYPE_ECC_KEY_PAIR(family)`
+
+           where ``family`` is a Weierstrass or Montgomery Elliptic curve family. That is, one of the following values:
+
+           *   ``PSA_ECC_FAMILY_SECT_XX``
+           *   ``PSA_ECC_FAMILY_SECP_XX``
+           *   `PSA_ECC_FAMILY_FRP`
+           *   `PSA_ECC_FAMILY_BRAINPOOL_P_R1`
+           *   `PSA_ECC_FAMILY_MONTGOMERY`
+
+.. macro:: PSA_ALG_ENCAPSULATION
+    :definition: /* specification-defined value */
+
+    .. summary::
+        Macro to build a combined algorithm that chains a key encapsulation with a key derivation.

```

Comment:
I don't think it will be easy to drive a combined algorithm here. The KDF requires additional inputs, and the specific input steps are KDF specific.

For the PSA Crypto API, I think it might be best to just provide an encapsulation API that results in the shared secret - for example, the value denoted `Z` in the SEC1 specification of ECIES. Derivation of the secret into encryption and authentication keys is then left to the application - or a higher-level ECIES function that wraps the PSA Crypto API calls.

This places these APIs at the same level as the key agreement APIs - typically an application must use key derivation to convert a possibly-biased shared secret into a usable encryption key.